pkg/k8s/profiles_test.go

```diff
@@ -488,6 +492,183 @@ func Test_PodSecurityProfile_Remove(t *testing.T) {
 	}
 }
 
+func Test_DNSPolicyProfile_Apply(t *testing.T) {
+	expectedPolicy := corev1.DNSDefault
+	p := Profile{DNSPolicy: &expectedPolicy}
+
+	basicDeployment := &appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: apiv1.PodTemplateSpec{
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{Name: "testfunc", Image: "alpine:latest"},
+					},
+				},
+			},
+		},
+	}
+
+	factory := mockFactory()
+	factory.ApplyProfile(p, basicDeployment)
+	result := basicDeployment.Spec.Template.Spec.DNSPolicy
+	if result == corev1.DNSClusterFirst {
+		t.Fatalf("expected %s, got DNSClusterFirst", expectedPolicy)
+	}
+	if expectedPolicy != result {
+		t.Fatalf("expected %s, got %v", expectedPolicy, result)
+	}
+}
+
+func Test_DNSPolicyProfile_Remove(t *testing.T) {
+	t.Run("remove matching DNS policy ", func(t *testing.T) {
+		expectedPolicy := corev1.DNSDefault
```

Comment:
this isn't really the `expectdPolicy`, i would call it `initialPolicy` or `currentPolicy`